# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsChildView.mm
# Commit: 2424d5c30267
# Full Hash: 2424d5c30267f67ab2b28119d75d89f0441b30e0
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2024-01-23 05:36:48
# Regressor Bug: 1833814
# File Overlap Count: 1
# Description:
#   Bug 1833814 - change nsIWidget's cursor logic, r=emilio,win-reviewers,mhowell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195549
# ==============================================================================

diff -r 94f46aad00df -r 2424d5c30267 widget/cocoa/nsChildView.mm
--- a/widget/cocoa/nsChildView.mm	Mon Jan 22 20:10:54 2024 +0000
+++ b/widget/cocoa/nsChildView.mm	Mon Jan 22 20:42:50 2024 +0000
@@ -625,9 +625,12 @@
 
   nsBaseWidget::SetCursor(aCursor);
 
-  if (NS_SUCCEEDED([[nsCursorManager sharedInstance]
-            setCustomCursor:aCursor
-          widgetScaleFactor:BackingScaleFactor()])) {
+  bool forceUpdate = mUpdateCursor;
+  mUpdateCursor = false;
+  if (mCustomCursorAllowed && NS_SUCCEEDED([[nsCursorManager sharedInstance]
+                                    setCustomCursor:aCursor
+                                  widgetScaleFactor:BackingScaleFactor()
+                                        forceUpdate:forceUpdate])) {
     return;
   }
 